consensus/istanbul/core/core.go

```diff
@@ -304,6 +308,12 @@ func (c *core) startNewRound(round *big.Int) {
 	var newView *istanbul.View
 	var roundChangeCertificate istanbul.RoundChangeCertificate
 	var request *istanbul.Request
+
+	var valSet istanbul.ValidatorSet
```

Comment:
How does the proposer get selected  if `valSet` is nil?  (e.g. when `c.current == nil`).

Reply:
The code is hard to follow since i didn't want to refactor this method, and it's quite a mess...

But is as follows:

1. When c.current is nil, then we enter on the first if (line 280); thus `roundChange = false`
2. Since roundChange is false, we enter the `else` on line 329, and there on line 337 valSet is set